src/olympia/addons/tasks.py

```diff
@@ -234,6 +236,21 @@ def recreate_theme_previews(addon_ids, **kw):
                                 quality=35,
                             )
                             preview.save()
+
+                            if old_thumbnail_path != preview.thumbnail_path:
+                                try:
+                                    log.info(
+                                        'Removing filename: %s for preview: %s'
+                                        % (old_thumbnail_path, preview.pk)
+                                    )
+                                    storage.delete(old_thumbnail_path)
```

Comment:
Although... this is kinda flawed.  The ES index will still be referring to the file we just deleted until the `index_addons` task completes 😒

Reply:
ideally we'd spin this out into a separate task and only start it after the `index_addons` task had finished, but celery task chaining is difficult to get right.

Reply:
We don't have to delete anything now, it can be done at a later date. It's probably safer, because there might be some places in the internet that still reference the old thumbnails - we don't want to keep them forever, but we can leave them some time to update, especially if that helps us too...

Reply:
https://github.com/mozilla/addons-server/pull/16710/commits/b2d3ee3ec96f39da492a1ec57b537fd9105628da

Reply:
for reference in case we change our minds:
```diff
--- a/src/olympia/addons/tasks.py
+++ b/src/olympia/addons/tasks.py
@@ -204,6 +204,7 @@ def recreate_theme_previews(addon_ids, **kw):
         }
         for render in amo.THEME_PREVIEW_RENDERINGS.values()
     }
+    thumbs_to_delete = []

     for version in versions:
         try:
@@ -238,18 +239,9 @@ def recreate_theme_previews(addon_ids, **kw):
                             preview.save()

                             if old_thumbnail_path != preview.thumbnail_path:
-                                try:
-                                    log.info(
-                                        'Removing filename: %s for preview: %s'
-                                        % (old_thumbnail_path, preview.pk)
-                                    )
-                                    storage.delete(old_thumbnail_path)
-                                except Exception as e:
-                                    log.error(
-                                        'Error deleting preview file (%s): %s',
-                                        old_thumbnail_path,
-                                        e,
-                                    )
+                                thumbs_to_delete.append(
+                                    (preview.id, old_thumbnail_path)
+                                )

                     continue
                 # else carry on with a full preview generation
@@ -261,6 +253,17 @@ def recreate_theme_previews(addon_ids, **kw):
         except IOError:
             pass
     index_addons.delay(addon_ids)
+    delete_preview_thumbs.delay(thumbs_to_delete)
+
+
+@task
+def delete_preview_thumbs(id_path_pairs):
+    for pk, path in id_path_pairs:
+        try:
+            log.info('Removing filename: %s for preview: %s' % (path, pk))
+            storage.delete(path)
+        except Exception as exc:
+            log.error('Error deleting preview file (%s): %s', path, exc)


 @task
diff --git a/src/olympia/addons/tests/test_tasks.py b/src/olympia/addons/tests/test_tasks.py
index fc62113eaf..288b9635ef 100644
--- a/src/olympia/addons/tests/test_tasks.py
+++ b/src/olympia/addons/tests/test_tasks.py
@@ -124,13 +124,14 @@ def test_create_missing_theme_previews(parse_addon_mock):
     assert VersionPreview.objects.count() == 3
     with mock.patch(f'{PATCH_PATH}.generate_static_theme_preview') as gstp, mock.patch(
         f'{PATCH_PATH}.resize_image'
-    ) as resize, mock.patch(f'{PATCH_PATH}.storage.delete') as storage:
+    ) as resize, mock.patch(f'{PATCH_PATH}.delete_preview_thumbs') as delete:
         old_firefox_thumb_path = firefox_preview.thumbnail_path
         recreate_theme_previews([theme.id], only_missing=True)
         assert gstp.call_count == 0  # not called
         assert resize.call_count == 2
-        assert storage.call_count == 1  # should only have been called for gif->png
-        assert storage.call_args == ((old_firefox_thumb_path,),)
+        assert delete.delay.call_count == 1  # Only will have been called for gif->png
+        assert delete.delay.call_args == (
+            ([(firefox_preview.id, old_firefox_thumb_path)],),)
         amo_preview.reload()
         assert amo_preview.thumbnail_dimensions == [720, 92]
         firefox_preview.reload()
diff --git a/src/olympia/lib/settings_base.py b/src/olympia/lib/settings_base.py
index 76c3c9a809..e24defceea 100644
--- a/src/olympia/lib/settings_base.py
+++ b/src/olympia/lib/settings_base.py
@@ -1118,12 +1118,13 @@ CELERY_TASK_ROUTES = {
     'olympia.addons.tasks.extract_colors_from_static_themes': {'queue': 'addons'},
     'olympia.devhub.tasks.pngcrush_existing_preview': {'queue': 'addons'},
     'olympia.devhub.tasks.pngcrush_existing_icons': {'queue': 'addons'},
-    'olympia.addons.tasks.recreate_theme_previews': {'queue': 'addons'},
     # Adhoc
     # A queue to be used for one-off tasks that could be resource intensive.
     'olympia.addons.tasks.hard_delete_extra_files': {'queue': 'adhoc'},
     'olympia.addons.tasks.hard_delete_legacy_versions': {'queue': 'adhoc'},
     'olympia.versions.tasks.delete_list_theme_previews': {'queue': 'adhoc'},
+    'olympia.addons.tasks.recreate_theme_previews': {'queue': 'adhoc'},
+    'olympia.addons.tasks.delete_preview_thumbs': {'queue': 'adhoc'},
     # Crons
     'olympia.addons.tasks.update_addon_average_daily_users': {'queue': 'cron'},
     'olympia.addons.tasks.update_addon_hotness': {'queue': 'cron'},
```